Bottom: 69a844b89d8aa5c51d33d674e587ec4703d1b0cc
Top:    9e043a25a903a5414a42acbdd93fa3e97dc63da0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-12-04 15:55:54 +0000

Make poly_Copy1 a static function internal to polygon1.c


---

diff --git a/src/polyarea.h b/src/polyarea.h
index c606077..1b8e9b3 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -140,7 +140,6 @@ BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
 void poly_M_Incl(POLYAREA **list, POLYAREA *a);
 
 BOOLp poly_Copy0(POLYAREA **dst, const POLYAREA *src);
-BOOLp poly_Copy1(POLYAREA  *dst, const POLYAREA *src);
 
 BOOLp poly_InclContour(POLYAREA * p, PLINE * c);
 BOOLp poly_ExclContour(POLYAREA * p, PLINE * c);
diff --git a/src/polygon1.c b/src/polygon1.c
index 2a8c669..1498462 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2789,20 +2789,7 @@ poly_CopyContour (PLINE ** dst, PLINE * src)
 /**********************************************************************/
 /* polygon routines */
 
-BOOLp
-poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
-{
-  *dst = NULL;
-  if (src != NULL)
-    *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
-    return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
-
-  return poly_Copy1 (*dst, src);
-}
-
-BOOLp
+static BOOLp
 poly_Copy1 (POLYAREA * dst, const POLYAREA * src)
 {
   PLINE *cur, **last = &dst->contours;
@@ -2813,13 +2800,27 @@ poly_Copy1 (POLYAREA * dst, const POLYAREA * src)
   for (cur = src->contours; cur != NULL; cur = cur->next)
     {
       if (!poly_CopyContour (last, cur))
-	return FALSE;
+        return FALSE;
       r_insert_entry (dst->contour_tree, (BoxType *) * last, 0);
       last = &(*last)->next;
     }
+
   return TRUE;
 }
 
+BOOLp
+poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
+{
+  *dst = NULL;
+  if (src != NULL)
+    *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
+  if (*dst == NULL)
+    return FALSE;
+  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
+
+  return poly_Copy1 (*dst, src);
+}
+
 void
 poly_M_Incl (POLYAREA ** list, POLYAREA * a)
 {
